tests/unit/backend/modules/user/user.service.test.ts

```diff
@@ -0,0 +1,206 @@
+import { ObjectID } from 'bson'
+import mongoose from 'mongoose'
+import { ImportMock } from 'ts-mock-imports'
+
+import getAdminVerificationModel from 'src/app/models/admin_verification.server.model'
+import * as UserService from 'src/app/modules/user/user.service'
+import * as OtpUtils from 'src/app/utils/otp'
+
+import dbHandler from '../../helpers/jest-db'
+
+const AdminVerification = getAdminVerificationModel(mongoose)
+
+describe('user.service', () => {
+  // Obtained from Twilio's
+  // https://www.twilio.com/blog/2018/04/twilio-test-credentials-magic-numbers.html
+  const MOCK_CONTACT = '+15005550006'
+  const MOCK_OTP = '123456'
+  const USER_ID = new ObjectID()
+
+  let defaultAgencyId: ObjectID
+
+  beforeAll(async () => {
+    await dbHandler.connect()
+
+    // Insert user into collections.
+    const { agency } = await dbHandler.insertFormCollectionReqs({
+      userId: USER_ID,
+    })
+
+    defaultAgencyId = agency._id
+  })
+  beforeEach(
+    async () =>
+      await dbHandler.clearCollection(
+        AdminVerification.collection.collectionName,
+      ),
+  )
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('createContactOtp', () => {
+    it('should create a new AdminVerification document and return otp', async () => {
+      // Arrange
+      // All calls to generateOtp will return MOCK_OTP.
+      ImportMock.mockFunction(OtpUtils, 'generateOtp', MOCK_OTP)
+      // Should have no documents prior to this.
+      await expect(AdminVerification.countDocuments()).resolves.toEqual(0)
+
+      // Act
+      const actualOtp = await UserService.createContactOtp(
+        USER_ID,
+        MOCK_CONTACT,
+      )
+
+      // Assert
+      expect(actualOtp).toEqual(MOCK_OTP)
+      // An AdminVerification document should have been created.
+      // Tests on the schema will be done in the schema's tests.
+      await expect(AdminVerification.countDocuments()).resolves.toEqual(1)
+    })
+
+    it('should throw error when userId is invalid', async () => {
+      // Arrange
+      const invalidUserId = new ObjectID()
+
+      // Act + Assert
+      await expect(
+        UserService.createContactOtp(invalidUserId, MOCK_CONTACT),
+      ).rejects.toThrowError('User id is invalid')
+    })
+  })
+
+  describe('verifyContactOtp', () => {
```

Comment:
worth adding tests for bcrypt error?

Reply:
No, `bcrypt` has their own tests. The error handling is already tested in `user.controller.spec` tests:

```ts
it('should return 500 when verifying contact throws non-ApplicationError', async () => {...})
```